### PR TITLE
Added sorting implementation programs in java

### PR DESCRIPTION
**Pull request by noobkid2411**
I created a folder with sorting in java as well as programs using recursion in c . @siddeshshewde Please look into my contributions and kindly approve and add a label so they count towards hacktober, if they seem valuable.Thanks!

### CODE REVIEW

File: C/Recursive Programs/add.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        Recursion: adding 1, n times and 
+        then at the end adding m to it
+    */
+    y = add(m, n-1) + 1;
+    return y;   // return the result
+}
+
+int main()
+{
+    printf("\n\n\t\tStudytonight - Best place to learn\n\n\n");
```
Comment (anonymous-sidhax) [inline C/Recursive Programs/add.c:24]: Looks like the code is copied from studytonight website. No point in copying bro :)

Anyways files look good. Will check all the files and merge it

**Review by anonymous-sidhax — CHANGES_REQUESTED** [review]
Please remove the unwanted part to keep the code clean.

I have added inline comments as well.

File: C/Recursive Programs/add.c (hunk 1 above)
Comment (noobkid2411) [reply C/Recursive Programs/add.c]: yeah sorry ,used to practice from there and directly copied from my old directory without edits.Really sorry again:)

File: Java/sorting/array_sort.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+
+edit
+play_arrow
+
+brightness_4
```
[the remark this answers is not in the document]
Comment (noobkid2411) [reply Java/sorting/array_sort.java]: edited

**Comment by noobkid2411** [comment]
sorry for the issues,I hope the codes are clean now.I'll be sure to review them well from now on. @anonymous-sidhax

**Review by siddeshshewde — APPROVED** [review]

**Pull request merged**